ext/drbbdup/drbbdup.c

```diff
@@ -959,6 +1113,163 @@ drbbdup_include_encoding(drbbdup_manager_t *manager, uintptr_t new_encoding)
     return false;
 }
 
+/****************************************************************************
+ * Dynamic case handling via flushing.
+ */
+
+static void
+drbbdup_prepare_redirect(dr_mcontext_t *mcontext, drbbdup_manager_t *manager,
+                         app_pc bb_pc)
+{
+    /* Restore flags and scratch reg to their original app values. */
+    if (!manager->are_flags_dead) {
+        reg_t val;
+        uint sahf;
+        reg_t newval = mcontext->xflags;
+        val = (reg_t)drbbdup_get_tls_raw_slot_val(DRBBDUP_FLAG_REG_SLOT);
+        sahf = (val & 0xff00) >> 8;
+        newval &= ~(EFLAGS_ARITH);
+        newval |= sahf;
+        if (TEST(1, val)) /* seto */
+            newval |= EFLAGS_OF;
```

Comment:
Hmm is there no API function in drreg or DR to do this flags merge for us?  TOFILE?

Reply:
I don't think there is one. IMO this should be added to DR and not drreg - just in case there are users who do not want to use drreg for flag restoration. In turn, drreg can then use the helper function as well.

Reply:
I filed #4194